library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala

```diff
@@ -1,46 +0,0 @@
-package scala.scalajs.runtime
-
-import scala.scalajs.js
-
-import StackTrace.JSStackTraceElem
-
-/** Information about the JavaScript environment Scala.js runs in.
- *
- *  Holds configuration for the Scala.js internals and should not be used
- *  directly (could be retrieved via [[runtime.environmentInfo]]).
- *
- *  This facade type serves as a documentation on what aspects of Scala.js can
- *  be influenced through environment options.
- *
- *  Upon startup, Scala.js checks whether the name <code>__ScalaJSEnv</code> is
- *  defined in its scope (and references an object). If so, it uses it as
- *  environment info.
- *  Missing, non-optional fields (according to this facade type) are initialized
- *  to default values, optional fields are kept as in the original object.
- *  Finally, [[js.Object.freeze]] is called on the object to avoid modification.
- *
- *  @groupname envInfo Scala.js environment configuration
- *  @groupprio envInfo 1
- */
-sealed trait EnvironmentInfo extends js.Object {
```

Comment:
I do not understand why you are getting rid of the typing here. It seems like this leads to very un-idiomatic Scala.js.

How about:

```
sealed trait EnviromentInfo // unchanged

object EnvironmentInfo {
  def envInfo: js.UndefOr[EnvironmentInfo] = {
    // ...
  }
}
```